SD_Assignment1/src/main/java/ro/utcn/aron/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,84 @@
+package ro.utcn.aron.jdbc;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Component;
+
+import ro.utcn.aron.api.QuestionRepository;
+import ro.utcn.aron.model.Question;
+
+@Component
+public class JdbcQuestionRepository implements QuestionRepository {
+
+	private  JdbcTemplate template; //final
+	
+	@Override
+	public Question save(Question question) {
+		if(question.getId() != 0) {
+			update(question);
+		} else {
+			int id = insert(question);
+			question.setId(id);
+		}
+		return null;
+	}
+
+	@Override
+	public Optional<Question> findByID(int id) {
+		List<Question> questions = template.query("SELECT * FROM question WHERE id = ?"+id, 
+				new Object[] { id },
+				(resultSet, i) -> 
+		new Question(resultSet.getInt("id"), resultSet.getString("title"), resultSet.getString("body"), resultSet.getString("tags")));
+		return questions.isEmpty() ? Optional.empty() : Optional.of(questions.get(0));
+	}
+
+	@Override
+	public void remove(Question question) {
+		template.update("DELETE * FROM question WHERE id = ?", question.getId());
+		
+	}
+
+	@Override
+	public List<Question> findAll() {
+		return template.query("SELECT * FROM question", (resultSet, i) -> 
+			new Question(resultSet.getInt("id"), resultSet.getString("title"), resultSet.getString("body"), resultSet.getString("tags")));
+	}
+	
+	private int insert(Question question) {
+		SimpleJdbcInsert insert = new SimpleJdbcInsert(template);
+		insert.setTableName("question");
+		insert.setGeneratedKeyName("id");
+		
+		Map<String, Object> data = new HashMap<>();
+		data.put("title", question.getTitle());
+		data.put("text", question.getBody());
+		
+		return insert.executeAndReturnKey(data).intValue();
+	}
+	
+	private void update(Question question) {
+		template.update("UPDATE question SET title = ?, text = ? WHERE id = ?  ",
+				question.getTitle(), question.getBody(), question.getId());
+	}
+
+	@Override
+	public Optional<List<Question>> filterByTitle(String title) {

```

Comment:
[Optional] Returning an Optional of List is usually not needed. Typically, you would return simply an empty list instead of an empty optional (and you would never return null instead of a list).